src/main/java/com/yahoo/identity/services/token/TokenCreateImpl.java

```diff
@@ -31,42 +25,36 @@ public TokenCreateImpl(@Nonnull KeyService keyService) {
     public TokenCreate setType(@Nonnull TypeEnum type) {
         switch (type) {
             case CRITICAL:
-                token.setTokenType(TokenType.CRITICAL);
+                this.token.setTokenType(TokenType.CRITICAL);
                 break;
             case STANDARD:
-                token.setTokenType(TokenType.STANDARD);
+                this.token.setTokenType(TokenType.STANDARD);
                 break;
             default:
-                throw new BadRequestException("Token type is not valid");
+                throw new IdentityException(IdentityError.INVALID_ARGUMENTS, "Unsupported token type.");
         }
         return this;
     }
 
     @Override
     @Nonnull
-    public TokenCreate setToken(@Nonnull String tokenStr) {
-        try {
-            Algorithm algorithm =
-                Algorithm.RSA256((RSAPublicKey) this.keyService.getPublicKey("token-public.pem", "RSA"),
-                                 (RSAPrivateKey) this.keyService.getPrivateKey("token-private.pem", "RSA"));
-            JWTVerifier verifier = JWT.require(algorithm).build();
-
-            DecodedJWT jwt = verifier.verify(tokenStr);
-
-            token.setSubject(jwt.getSubject());
-            token.setIssueTime(jwt.getIssuedAt().toInstant());
-            token.setExpireTime(jwt.getExpiresAt().toInstant());
-
-        } catch (JWTVerificationException e) {
-            throw new IdentityException(IdentityError.INVALID_CREDENTIAL, "JWT verification does not succeed.");
+    public void initToken(@Nonnull String username) {
+        this.token.setSubject(username);
+        this.token.setIssueTime(Instant.now());
```

Comment:
Change to builder pattern will be easy to do. However, `Account` is immutable because we put all the setters into `AccountModel`, where all the data are stored.